data_collection/gazette/spiders/al/al_maragogi.py

```diff
@@ -0,0 +1,91 @@
+import locale
+from datetime import date, datetime
+from urllib.parse import ParseResult, parse_qs, urlencode, urlparse
+
+import scrapy
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class AlMaragogiSpider(BaseGazetteSpider):
+    name = "al_maragogi"
+    TERRITORY_ID = "2704500"
+    allowed_domains = ["diario.maragogi.al.gov.br"]
+    start_date = date(2024, 4, 17)
+    start_urls = ["https://diario.maragogi.al.gov.br"]
+
+    def start_requests(self):
+        params = {
+            "BuscaSearch[data_inicio]": self.start_date.strftime("%Y-%m-%d"),
+            "BuscaSearch[data_fim]": self.end_date.strftime("%Y-%m-%d"),
+            "BuscaSearch[sort]": "data_new",
+            "BuscaSearch[modulo]": "diario-oficial",
+            "per-page": 30,
+        }
+        url = f"{self.start_urls[0]}/busca?{urlencode(params, encoding='utf-8')}"
+        yield scrapy.Request(
+            url,
+        )
```

Comment:
Três sugestões aqui:
- Aproveitar uma funcionalidade da `FormRequest` quando usada com `method=GET` de colocar o `formdata` na querystring da URL
- `per-page` pode ser removido pois não influencia no resultado (e infelizmente não pode ser regulado para aumentar o número de resultado por página)
- Fazer um método para gerar os params já que em outro método também está sendo criado praticamente igual
```suggestion
    def start_requests(self):
        yield scrapy.FormRequest(
            url=f"{self.BASE_URL}/busca",
            method="GET",
            formdata=self._build_formdata(),
        )

    def _build_formdata(self, page=1):
        return {
            "BuscaSearch[data_inicio]": self.start_date.strftime("%Y-%m-%d"),
            "BuscaSearch[data_fim]": self.end_date.strftime("%Y-%m-%d"),
            "BuscaSearch[sort]": "data_new",
            "BuscaSearch[modulo]": "diario-oficial",
            "page": str(page),
        }

```